src/cargo/commands.rs

```diff
@@ -0,0 +1,168 @@
+
+use cargo::command_bridge::CommandBridge;
+use std::path::{Path, PathBuf};
+use std::process::Stdio;
+
+use error::*;
+
+pub fn retrieve_metadata(manifest_path: &Path) -> CommandBridge {
+    let mut command = CommandBridge::new("cargo");
+    command.arg("metadata")
```

Comment:
I'd prefer it to be a builder pattern stylistically so you can do:

```rust
CommandBridge::new("cargo")
    .arg("metadata")
    .arg("--manifest-path")
    .arg(manifest_path)
    .arg("--no-deps")
    .arg("--format-version")
    .arg("1")
```

If you change the function `arg` to  `arg(self) -> Self` this enables that pattern. Builder patterns are great and this is essentially what you're doing here! :D

Reply:
Yes, it would be more canonic. I'll change

Reply:
:D